### PR TITLE
Merge from gui-drives dev branch

### PR DESCRIPTION
**Pull request by qu-gg**
Added functionality and correct logic parsing for the GUI elements of the j- and m- drives.

Refactored some documentation in spacecraft.py for uniformity.

Updated spacecraft::set_tonnage to properly alter cargo and cost when switching the tonnage. May need more testing, but related objects seems to be adjusting correctly.

Note: at the moment, all error messages are printed out to the terminal window. Direction could be a small box within the GUI that handles these stdout statements.

### CODE REVIEW

File: imperium/models/spacecraft.py [hunk 1]
```diff
@@ -76,16 +76,29 @@ def __init__(self, hull_tonnage):
     def set_tonnage(self, new_tonnage):
         """
         Sets the tonnage of an existing Spacecraft
-
         :param new_tonnage: The tonnage to update to
         """
+        tonnage_cost_data = get_file_data("hull_data.json")
+        new_cost = 0
+        for key in tonnage_cost_data:
```
Comment (Milkshak3s) [inline imperium/models/spacecraft.py]: Iterate through values, save lookups

**Comment by qu-gg** [comment]
Changes made, requesting rereview

File: imperium/models/spacecraft.py [hunk 2]
```diff
@@ -94,66 +106,80 @@ def set_fuel(self, new_fuel):
...
+            # if drive already in place, replace stats
+            if self.jdrive is not None:
+                self.cost_total = self.cost_total - self.jdrive.cost
+                self.cargo = self.cargo + self.jdrive.tonnage
 
-        # assign object to ship, update cost & tonnage
-        self.jdrive = new_jdrive
-        self.cost_total = self.cost_total + new_jdrive.cost
-        self.cargo = self.cargo - new_jdrive.tonnage
-        self.performance_by_volume("jdrive", drive_type)
```
Comment (qu-gg) [inline imperium/models/spacecraft.py]: This is kinda garbo consider calling a different function

Comment (qu-gg) [reply imperium/models/spacecraft.py]: Oh you're actually dumb

**Pull request merged**